addons/pkg-maven/ftests/src/main/java/org/commonjava/indy/pkg/maven/content/MavenContentAdvisor.java

```diff
@@ -0,0 +1,59 @@
+/**
+ * Copyright (C) 2013 Red Hat, Inc. (jdcasey@commonjava.org)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.commonjava.indy.pkg.maven.content;
+
+import org.commonjava.indy.spi.pkg.ContentAdvisor;
+import org.commonjava.indy.spi.pkg.ContentQuality;
+import org.commonjava.maven.atlas.ident.util.ArtifactPathInfo;
+import org.commonjava.maven.galley.model.SpecialPathInfo;
+import org.commonjava.maven.galley.spi.io.SpecialPathManager;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+/**
+ */
+@ApplicationScoped
+public class MavenContentAdvisor
+        implements ContentAdvisor
+{
+    @Inject
+    private SpecialPathManager specialPathManager;
+
+    @Override
+    public ContentQuality getContentQuality( String path )
+    {
+        final ArtifactPathInfo pathInfo = ArtifactPathInfo.parse( path );
+        if ( pathInfo != null )
+        {
+            if ( pathInfo.isSnapshot() )
+            {
+                return ContentQuality.SNAPSHOT;
+            }
+
+            final SpecialPathInfo info = specialPathManager.getSpecialPathInfo( path );
+            if ( info != null && info.isMetadata() )
+            {
+                return ContentQuality.METADATA;
+            }
+
+            //FIXME: if any flag to identify a path is a RELEASE quality?
+            return ContentQuality.RELEASE;
```

Comment:
You're right here, it's just a process of elimination. If it's a maven artifact and it's not a metadata or snapshot file, it's a release file.

Though, one thing I'm not sure of: We might want RELEASE to be the same as a null result here. It's possible the path pattern might match a Maven release-quality artifact, yet not really be a maven artifact. This is a LOT more likely than a false positive from the other two quality levels, since those things look for much more specific parts of the path.

For instance:
```
   /some/path/to/a/non-maven/one/non-maven-one.txt
```

This would parse as an artifact with:
* groupId = some.path.to.a
* artifactId = non-maven
* version = one
* type = txt
* classifier = [null]

Yet it could be a PyPI path or something for all we know right now. I think it might be safest to assume that RELEASE is the quality level of things for which no ContentAdvisor has specified a different level.